# This script provides answers to interactive ./configure and replaces paths for GCC and Binutils
# @author:
#      Marcel Schoengens (CSCS)
#      Guilherme Peretti-Pezzi (CSCS)
#      Victor Holanda Rusu (CSCS)
diff --git a/configure-cscs.sh b/configure-cscs.sh
new file mode 100644
index 0000000..799fdbf
--- /dev/null
+++ b/configure-cscs.sh
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+sed -i s\@'$EBROOTGCCCORE'@"$EBROOTGCCCORE"@g third_party/gpus/crosstool/CROSSTOOL.tpl
+sed -i s\@'$EBROOTBINUTILS'@"$EBROOTBINUTILS"@g third_party/gpus/crosstool/CROSSTOOL.tpl
+sed -i s\@'$EBROOTGCCCORE'@"$EBROOTGCCCORE"@g third_party/gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl
+
+CONF_IN="configure.in"
+
+echo $EBROOTPYTHON"/bin/python" > $CONF_IN
+echo "-march=native" >> $CONF_IN
+echo "n" >> $CONF_IN
+echo "n" >> $CONF_IN
+echo "n" >> $CONF_IN
+echo "n" >> $CONF_IN
+echo $EBROOTPYTHON"/lib/python3.5/site-packages" >> $CONF_IN
+echo "n" >> $CONF_IN
+echo "n" >> $CONF_IN
+
+cat $CONF_IN
+echo "---"
+./configure < $CONF_IN
+
+exit
diff --git a/tensorflow/core/platform/default/build_config.bzl b/tensorflow/core/platform/default/build_config.bzl
index 48ef8df..be831d5 100644
--- a/tensorflow/core/platform/default/build_config.bzl
+++ b/tensorflow/core/platform/default/build_config.bzl
@@ -8,7 +8,7 @@ load("//tensorflow:tensorflow.bzl", "if_not_mobile")
 WITH_GCP_SUPPORT = False
 WITH_HDFS_SUPPORT = False
 WITH_XLA_SUPPORT = False
-WITH_JEMALLOC = True
+WITH_JEMALLOC = False

 # Appends a suffix to a list of deps.
 def tf_deps(deps, suffix):
diff -ru tensorflow-1.0.0-orig/third_party/gpus/crosstool/CROSSTOOL.tpl tensorflow-1.0.0/third_party/gpus/crosstool/CROSSTOOL.tpl
--- tensorflow-1.0.0-orig/third_party/gpus/crosstool/CROSSTOOL.tpl	2017-02-11 07:33:43.000000000 +0100
+++ tensorflow-1.0.0/third_party/gpus/crosstool/CROSSTOOL.tpl	2017-03-28 19:17:42.743394544 +0200
@@ -42,10 +42,10 @@
   target_system_name: "local"
   toolchain_identifier: "local_linux"
 
-  tool_path { name: "ar" path: "/usr/bin/ar" }
-  tool_path { name: "compat-ld" path: "/usr/bin/ld" }
-  tool_path { name: "cpp" path: "/usr/bin/cpp" }
-  tool_path { name: "dwp" path: "/usr/bin/dwp" }
+  tool_path { name: "ar" path: "$EBROOTBINUTILS/bin/ar" }
+  tool_path { name: "compat-ld" path: "$EBROOTBINUTILS/bin/ld" }
+  tool_path { name: "cpp" path: "$EBROOTGCCCORE/bin/cpp" }
+  tool_path { name: "dwp" path: "$EBROOTBINUTILS/bin/dwp" }
   # As part of the TensorFlow release, we place some cuda-related compilation
   # files in @local_config_cuda//crosstool/clang/bin, and this relative
   # path, combined with the rest of our Bazel configuration causes our
@@ -56,21 +56,23 @@
   cxx_flag: "-std=c++11"
   linker_flag: "-Wl,-no-as-needed"
   linker_flag: "-lstdc++"
-  linker_flag: "-B/usr/bin/"
+  linker_flag: "-B$EBROOTBINUTILS/bin/"
+#  linker_flag: "-L$EBROOTGCCCORE/lib64/"
+  linker_flag: "-Wl,-rpath, $EBROOTGCCCORE/lib64/"
 
 %{gcc_host_compiler_includes}
-  tool_path { name: "gcov" path: "/usr/bin/gcov" }
+  tool_path { name: "gcov" path: "$EBROOTGCCCORE/bin/gcov" }
 
   # C(++) compiles invoke the compiler (as that is the one knowing where
   # to find libraries), but we provide LD so other rules can invoke the linker.
-  tool_path { name: "ld" path: "/usr/bin/ld" }
+  tool_path { name: "ld" path: "$EBROOTBINUTILS/bin/ld" }
 
-  tool_path { name: "nm" path: "/usr/bin/nm" }
-  tool_path { name: "objcopy" path: "/usr/bin/objcopy" }
+  tool_path { name: "nm" path: "$EBROOTBINUTILS/bin/nm" }
+  tool_path { name: "objcopy" path: "$EBROOTBINUTILS/bin/objcopy" }
   objcopy_embed_flag: "-I"
   objcopy_embed_flag: "binary"
-  tool_path { name: "objdump" path: "/usr/bin/objdump" }
-  tool_path { name: "strip" path: "/usr/bin/strip" }
+  tool_path { name: "objdump" path: "$EBROOTBINUTILS/bin/objdump" }
+  tool_path { name: "strip" path: "$EBROOTBINUTILS/bin/strip" }
 
   # Anticipated future default.
   unfiltered_cxx_flag: "-no-canonical-prefixes"
diff -ru tensorflow-1.0.0-orig/third_party/gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl tensorflow-1.0.0/third_party/gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl
--- tensorflow-1.0.0-orig/third_party/gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl	2017-02-11 07:33:43.000000000 +0100
+++ tensorflow-1.0.0/third_party/gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl	2017-03-28 19:18:57.039063443 +0200
@@ -51,7 +51,7 @@
 
 CURRENT_DIR = os.path.dirname(sys.argv[0])
 NVCC_PATH = CURRENT_DIR + '/../../../cuda/bin/nvcc'
-LLVM_HOST_COMPILER_PATH = ('/usr/bin/gcc')
+LLVM_HOST_COMPILER_PATH = ('$EBROOTGCCCORE/bin/gcc')
 PREFIX_DIR = os.path.dirname(GCC_HOST_COMPILER_PATH)
 NVCC_VERSION = '%{cuda_version}'
 
